### ISSUE TITLE
Refine TAD non-coding assignment model

### ISSUE THREAD

**Issue by visze** [exomiser/Exomiser#132]
GeneReassigner works only on variants in "regulatory regions". But it does not reassign other non-coding variants in a TAD. Especially intronic variants. These variants should be reassigned:
- `CODING_TRANSCRIPT_INTRON_VARIANT`
- `CONSERVED_INTERGENIC_VARIANT`
- `CONSERVED_INTRON_VARIANT`
- `DOWNSTREAM_GENE_VARIANT`
- `INTERGENIC_REGION`
- `INTERGENIC_VARIANT`
- `INTRAGENIC_VARIANT`
- `INTRON_VARIANT`
- `NON_CODING_TRANSCRIPT_INTRON_VARIANT`
- `REGULATORY_REGION_VARIANT`
- `TF_BINDING_SITE_VARIANT`
- `UPSTREAM_GENE_VARIANT`

**Comment by julesjacobsen** [comment]
This is true, but before variants are run through the GeneReassigner they are run through the ```setRegulatoryRegionVariantEffect``` function in ```AbstractAnalysisRunner``` (line 233) this will reassign the variant effect from INTERGENIC_VARIANT or UPSTREAM_GENE_VARIANT to REGULATORY_REGION_VARIANT. The reassignment is based on whether ot not the variant falls inside a known FANTOM enhancer.

**Comment by visze** [comment]
Yes I know. But the fantom things is crap. Why we should create a score (remm) over the complete genome but only allowing the variants to be at obvious sites. So there is no need of the FANTOM thing.

E.g. The overlap of fantom with a specific limp enhancer track is <1%. This shows that limiting the "regulatory" variants to FAMTOM is useless. It is better to limit the variants to remm >0.5

I implemented a draft version where all variants are reassigned that fall into the non-coding varianteffects. it looks much better on our genomes now, because candidates, found by other approaches, are now present.

**Comment by julesjacobsen** [comment]
So yeah, this is somewhat coupled and not obvious. We're really trying to reassign variants in known enhancer regions to the best phenotypic match gene for all genes in that TAD. However I don't think we want to be too over-zealous with this as we will end up over-assigning variants to genes they have nothing to do with. That said, we could probably widen the reassignment to variants in the  DOWNSTREAM_GENE_VARIANT and CONSERVED_INTERGENIC_VARIANT too. @damiansm what were your criteria for only using the INTERGENIC_VARIANT and UPSTREAM_GENE_VARIANT effects?

**Comment by visze** [comment]
Well the thing is that I try to convince our physicians to use Genomiser on our genomes. But they say, and I agree with them, that any variant that somehow regulatory (conserved, acetylathed,... => large remm score) in a TAD with a gene matching the phenotype is of interest.

So right now do not want to use genomiser because of that issue. Right now they are looking at de-novos only and searching for a good gene in the same TAD.

The "overassignment" works very well on my test. So why should we do not do it?

DOWNSTREAM_GENE_VARIANT (it can also be upstream) is maybe to close to the gene.

**Comment by damiansm** [comment]
Think we should have a major discuss before we start messing with this
filter. It is not just FANTOM, it is Ensembl regulatory build as well that
we use. The geneReassigner is doing a different task as Jules described.

Max - sound like you are suggesting filtering variants by ReMM score rather
than regulatory features. Before we change things I would like to see some
hard evidence e.g. benchmarking, rather than just opinions. Suspect keeping
all variants with RemM > 0.5 is not going to filter that much and leave too
many candidates. FANTOM+Ensembl regulatory features did a pretty good job
of filtering and prioritisiing the curated variants.

**Comment by visze** [comment]
Damian you are right. It is not only fantom. Maybe we can have two different options. A "stringend" (as is it right now, maybe adding intronic and downstream variants to check if they are in a regulatory feature. I do not understand why did you exclude them. ) and a "loose" one (do not use fantom+ensembl).

Right now we did not solve any of our genomes with the Genomiser (ok the GLI2 case, but because of a missense variant). In the rest of the cases the resulting genes/variants do not match at all to the phenotype (exomiser score only around 0.5 -0.6). So it would be nice to to make a wider search.

**Comment by damiansm** [comment]
My brain is a bit fuzzy on all this and the code is a bit convoluted on how
all this works but you could just remove the regulatoryFeatureFilter in the
yml config?

Note the regulatoryFeatureFilter is not quite as simple as remove any
non-coding variant if it is not in fantom/ensembl. It also keeps ALL
variants within 25kb of the gene. Or was it 50kb? This feature may not be
obvious in the actual method that is doing the filtering!

**Comment by damiansm** [comment]
Trying to summarise all this.
Current nc variant behaviour is:
(1) Any variants found in the **regulatory_regions db table** of known regulatory regions from FANTOM and Ensembl regulatory build **AND an effect of INTERGENIC_VARIANT or UPSTREAM_GENE_VARIANT** get the effect changed to **REGULATORY_REGION_VARIANT**. This is to stop them getting removed in step 3 (? if Jannovar ever assigns this effect - I don't think so)
(2) Any variants with an effect of **REGULATORY_REGION_VARIANT** get **reassigned** to the gene in the TAD with the best pheno score
(3) The regulatoryFeature filter **removes** any variant with an effect of **INTERGENIC_VARIANT or UPSTREAM_GENE_VARIANT** **AND >= 20kb away from gene**

**Comment by julesjacobsen** [comment]
Hang on what's happening here? #142, #143 and this one are all no longer issues or is there a new issue relating to this?

**Comment by damiansm** [comment]
I created a one new issue merging all the info/discussions in these. It was
getting too messy

**Comment by damiansm** [comment]
see  #219 for this now

**Comment by julesjacobsen** [comment]
Ah, good. Even the simplification was confusing!
